src/main/java/com/esewa/usermanagement/service/AdminServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+package com.esewa.usermanagement.service;
+
+import com.esewa.usermanagement.constants.ExceptionMessages;
+import com.esewa.usermanagement.entity.RegistrationLog;
+import com.esewa.usermanagement.entity.User;
+import com.esewa.usermanagement.enums.RoleType;
+import com.esewa.usermanagement.exceptions.UserRegistrationFailedException;
+import com.esewa.usermanagement.repository.UserRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+@Slf4j
+@Service
+public class AdminServiceImpl implements AdminService{
+
+    private final UserService userService;
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final LogService logService;
+
+    public AdminServiceImpl (UserService userService, UserRepository userRepository, PasswordEncoder passwordEncoder, LogService logService) {
+        this.userService = userService;
+        this.userRepository = userRepository;
+        this.passwordEncoder = passwordEncoder;
+        this.logService = logService;
+        this.createAdminUser();
+    }
+
+    @Override
+    public User registerUser(User user) {
+        try {
+            CompletableFuture<User> userFuture = userService.registerUser(user);
+            return userFuture.get();
+        } catch (Exception exception) {
+            log.info(exception.getMessage());
+            throw new UserRegistrationFailedException(ExceptionMessages.USER_REGISTRATION_FAILED_MESSAGE);
+        }
+    }
+
+    @Override
+    public List<User> registerMultipleUsers(List<User> userList) {
+
+        List<CompletableFuture<User>> asyncUserRegistrationList = new ArrayList<>();
+        List<User> registeredUserList = new ArrayList<>();
+        for (User user : userList) {
+            CompletableFuture<User> future = userService.registerUser(user);
+            asyncUserRegistrationList.add(future);
+        }
+        for (CompletableFuture<User> future : asyncUserRegistrationList) {
+            future.handle((user, ex) -> {
```

Comment:
Is it possible to move this handler to  userService.registerUser? Or atleast can you move this to line 52 so that we don't have to loop again.